src/txn/txn_rollback_to_stable.c

```diff
@@ -346,17 +346,17 @@ __rollback_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_REF *ref, WT_PAGE *page
 
         /*
          * Do not include history store updates greater than on-disk data store version to construct
-         * a full update to restore. Include the most recent updates than the on-disk version
-         * shouldn't be problem as the on-disk version in history store is always a full update. It
-         * is better to not to include those updates as it unnecessarily increases the rollback to
-         * stable time.
+         * a full update to restore except when the on-disk update is prepared. Include the most
```

Comment:
Do we fix out of timestamps if we have a prepared update with out-of-order timestamp? E.g., prepared@20, U1@30, U1@20, will be write U1@20 and U1@20 to history store?

Reply:
No. we don't do that. It is because prepared updates are not yet committed. The added test case demonstrates that we write Prepared@20 to the data store and U1@30 to the history store.